qa/integration-tests/src/test/java/io/camunda/it/utils/CamundaMultiDBExtension.java

```diff
@@ -0,0 +1,238 @@
+ *
+ * <pre>{@code
+ * @MultiDbTest
+ * final class MyMultiDbTest {
+ *
+ *   private CamundaClient client;
+ *
+ *   @Test
+ *   void shouldMakeUseOfClient() {
+ *     // given
+ *     // ... set up
+ *
+ *     // when
+ *     topology = c.newTopologyRequest().send().join();
+ *
+ *     // then
+ *     assertThat(topology.getClusterSize()).isEqualTo(1);
+ *   }
+ * }</pre>
+ *
+ * <p>There are more complex scenarios that might need to start respective TestApplication externally.
+ * For such cases the extension can be used via:
+ * <pre>{@code
+ * @RegisterExtension
+ * public final CamundaMultiDBExtension extension =
+ *    new CamundaMultiDBExtension(new TestStandaloneBroker());
+ * }</pre>
+ *
+ *<p>The extension will take care of the life cycle of the {@link TestStandaloneApplication}, which
+ * means configuring the detected database (this includes Operate, Tasklist, Broker properties and
+ * exporter), starting the application, and tearing down at the end.
+ *
+ * <p>See {@link TestStandaloneApplication} for more details.
+ */
+public class CamundaMultiDBExtension
+    implements AfterAllCallback, BeforeAllCallback, ParameterResolver {
+
+  public static final String PROP_CAMUNDA_IT_DATABASE_TYPE =
+      "test.integration.camunda.database.type";
+  public static final String DEFAULT_ES_URL = "http://localhost:9200";
+  public static final String DEFAULT_OS_URL = "http://localhost:9200";
+  public static final String DEFAULT_OS_ADMIN_USER = "admin";
+  public static final String DEFAULT_OS_ADMIN_PW = "yourStrongPassword123!";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CamundaMultiDBExtension.class);
+  private final DatabaseType databaseType;
+  private final List<AutoCloseable> closeables = new ArrayList<>();
+  private final TestStandaloneApplication<?> testApplication;
+  private String testPrefix;
+  private final MultiDbConfigurator multiDbConfigurator;
+
+  public CamundaMultiDBExtension() {
+    this(new TestSimpleCamundaApplication());
+    closeables.add(testApplication);
+    testApplication
+        .withBrokerConfig(cfg -> cfg.getGateway().setEnable(true))
+        .withExporter(
+            "recordingExporter", cfg -> cfg.setClassName(RecordingExporter.class.getName()));
+  }
+
+  public CamundaMultiDBExtension(final TestStandaloneApplication testApplication) {
+    this.testApplication = testApplication;
+    multiDbConfigurator = new MultiDbConfigurator(testApplication);
+    // resolve active database and exporter type
+    final String property = System.getProperty(PROP_CAMUNDA_IT_DATABASE_TYPE);
+    databaseType =
+        property == null ? DatabaseType.LOCAL : DatabaseType.valueOf(property.toUpperCase());
+  }
+
+  @Override
+  public void beforeAll(final ExtensionContext context) {
+    LOGGER.info("Starting up Camunda instance, with {}", databaseType);
+    final Class<?> testClass = context.getRequiredTestClass();
+    testPrefix = testClass.getSimpleName().toLowerCase();
```

Comment:
:question: Is simple name like, given a class `io.camunda.it.qa.MyTest`, just `MyTest`? Because we could have collisions without package names. :wrench: So using the fully qualified name would be a better option here, just in case.